### PR TITLE
Adds monster flags "BASIC_INVISIBILITY" and "FULL_INVISIBILITY"

### PR DESCRIPTION
**Pull request by NotFuji**
Two flags for invisible monsters:
BASIC_INVISIBILITY : Monster is invisible if greater than 5 tiles away
FULL_INVISIBILITY : Monster is only seen when on an adjacent tile

Mod with sample creatures: https://github.com/NotFuji/FujiMon

### CODE REVIEW

File: src/monstergenerator.cpp [hunk 1]
```diff
@@ -442,6 +442,8 @@ void MonsterGenerator::init_flags()
     flag_map["GROUP_MORALE"] = MF_GROUP_MORALE;
     flag_map["INTERIOR_AMMO"] = MF_INTERIOR_AMMO;
     flag_map["NIGHT_INVISIBILITY"] = MF_NIGHT_INVISIBILITY;
+	flag_map["BASIC_INVISIBILITY"] = MF_BASIC_INVISIBILITY;
```
Comment (Night-Pryanik) [inline src/monstergenerator.cpp]: Please replace tabs with 4 spaces.

File: src/mtype.h [hunk 2]
```diff
@@ -152,6 +152,8 @@ enum m_flag : int {
     MF_CLIMBS,              // Monsters that can climb certain terrain and furniture
     MF_PUSH_MON,            // Monsters that can push creatures out of their way
     MF_NIGHT_INVISIBILITY,  // Monsters that are invisible in poor light conditions
+	MF_BASIC_INVISIBILITY,  // Monsters that are invisible at distances greater than 5 tiles
+	MF_FULL_INVISIBILITY,   // Monsters that are invisible
```
Comment (Night-Pryanik) [inline src/mtype.h]: The same here.

**Comment by NotFuji** [comment]
New flags are intended to work with mods, and add consistency, since there is a NIGHT_INVISIBILITY flag, but no accompanying flag for daytime

**Comment by NotFuji** [comment]
Mod with sample creatures: https://github.com/NotFuji/FujiMon
Monsters "bloodsucker" and "black bloodsucker" use BASIC_INVISIBILITY and FULL_INVISIBILITY respectively.

**Comment by Night-Pryanik** [comment]
PRs that only add functionality to the mods not shipped with the game most likely won't get merged. You need to add your sample creatures to the vanilla or, if you think they don't make much sense to the vanilla, propose your mod with this PR.

**Comment by thethunderhawk** [comment]
Could be a perfect opportunity to add a new nether creature to spice things up. Give it a natural noise to give people a chance to react to it, make it a rare lab spawn, and put it out there. It’d add some spice to the game and if it were shoggoth-level rare I doubt anyone would object to it, if nothing else it light spice up those lab prisoner rooms. Plus, I look forward to the uses other modders might find for this.

**Comment by ZhilkinSerg** [comment]
Please add example mod to PR and resolve conflicts and I will merge test it.

**Pull request closed without merging**